virttest/utils_misc.py

```diff
@@ -3517,13 +3518,16 @@ def get_pci_group_by_id(pci_id, device_type=""):
     return []
 
 
-def get_pci_vendor_device(pci_id):
+def get_pci_vendor_device(pci_id, session=None):
     """
     Get vendor and device number by pci id.
 
+    :param pci_id: pci id of a device
+    :param session: ShellSession object of VM or remote host
     :return: a 'vendor device' list include all matched devices
     """
-    matched_pci = process.run("lspci -n -s %s" % pci_id, ignore_status=True).stdout_text
+    cmd = "lspci -n -s %s" % pci_id
+    _, matched_pci = cmd_status_output(cmd, shell=True, session=session)
```

Comment:
As per the above comments, I should add another branch `session.cmd_status_output(cmd)` to support the guest session situation to keep the original behaviors(In some cases, it will raise the error, like FileNotFoundError).
```suggestion
   if session:
        _, matched_pci = session.cmd_status_output(cmd)
   else:
       matched_pci = process.run("lspci -n -s %s" % pci_id, ignore_status=True).stdout_text

Reply:
I have updated and test is passed. please help review.

Reply:
@PaulYuuu, here is the history why not using cmd_status_output.

Reply:
we don't want the status, and we know we have a replacement `cmd_output`, keeping the bad method is not good.

Reply:
@YongxueHong any comments here ?

Reply:
I have them with cmd_output. please help review.

1-Host_RHEL.m9.u5.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.9.3.0.x86_64.io-github-autotest-qemu.pktgen_perf.vp_vdpa_test_vm.q35 Finshed 2024-06-17 02:55:30 2024-06-17 03:01:06 PASS 335.781488

Reply:
I have updated them with cmd_output. please help review.